File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-3.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        l = 0
        r = len(nums) - 1

        while l < r:
            if nums[l] == nums[l + 1]:
                nums.pop(l)
                r -= 1
            elif nums[r] == nums[r - 1]:
                nums.pop(r)
                r -= 1
            else:
                l += 1
                r -= 1
        
        return len(nums)

        


        

File: Data Structures & Algorithms/duplicate-integer/submission-3.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        set<int> seen;

        for (const int x : nums) {
            if (seen.count(x)) {
                return true;
            }

            seen.insert(x);
        }

        return false;
    }
};